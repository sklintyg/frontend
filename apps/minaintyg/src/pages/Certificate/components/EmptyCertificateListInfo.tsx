import { IDSAlert } from 'ids-react-ts'

export function EmptyCertificateListInfo({ total }: { total: number }) {
  if (total > 0) {
    return (
      <IDSAlert role="alert" headline="Din filtrering gav inga träffar">
        <p>Testa att ändra de filter du valt. Du kan också klicka på rensa filter för att se alla dina intyg.</p>
      </IDSAlert>
    )
  }
  return (
    <IDSAlert role="alert" headline="Du har inga intyg">
      <p>
        I Intyg visas endast intyg som utfärdats digitalt till dig. Intyg har ingen ombudsfunktion och du kan inte ladda upp egna intyg.
      </p>
      <p>Om du saknar ett intyg, vänd dig till din mottagning.</p>
    </IDSAlert>
  )
}
