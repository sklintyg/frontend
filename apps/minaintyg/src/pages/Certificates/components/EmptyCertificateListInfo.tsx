import { IDSAlert } from '@frontend/ids-react-ts'

export function EmptyCertificateListInfo() {
  return (
    <IDSAlert headline="Du har inga intyg">
      <p>
        I Intyg visas endast intyg som utfärdats digitalt till dig. Intyg har ingen ombudsfunktion och du kan inte ladda upp egna intyg.
      </p>
      <p>Om du saknar ett intyg, vänd dig till din mottagning.</p>
    </IDSAlert>
  )
}
