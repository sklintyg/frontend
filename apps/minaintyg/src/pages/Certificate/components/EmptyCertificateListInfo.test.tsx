import { render } from '@testing-library/react'
import { EmptyCertificateListInfo } from './EmptyCertificateListInfo'

it('Should display information if total above 0', () => {
  const { container } = render(<EmptyCertificateListInfo total={1} />)
  expect(container).toMatchSnapshot()
})

it('Should display information if total is 0', () => {
  const { container } = render(<EmptyCertificateListInfo total={0} />)
  expect(container).toMatchSnapshot()
})
