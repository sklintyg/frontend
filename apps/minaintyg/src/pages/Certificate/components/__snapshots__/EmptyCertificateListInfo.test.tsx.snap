// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should display information if total above 0 1`] = `
<div>
  <div
    class="ids-alert ids-alert--info"
    role="alert"
  >
    <div
      class="ids-alert__header"
    >
      <div
        class="ids-alert__icon_and_text"
      >
        <span
          class="ids-alert__state-icon"
          title=""
        />
        <div
          class="ids-alert__headline"
        >
          Din filtrering gav inga träffar
        </div>
      </div>
    </div>
    <div
      class="ids-alert__content"
    >
      <p>
        Testa att ändra de filter du valt. Du kan också klicka på rensa filter för att se alla dina intyg.
      </p>
    </div>
  </div>
</div>
`;

exports[`Should display information if total is 0 1`] = `
<div>
  <div
    class="ids-alert ids-alert--info"
    role="alert"
  >
    <div
      class="ids-alert__header"
    >
      <div
        class="ids-alert__icon_and_text"
      >
        <span
          class="ids-alert__state-icon"
          title=""
        />
        <div
          class="ids-alert__headline"
        >
          Du har inga intyg
        </div>
      </div>
    </div>
    <div
      class="ids-alert__content"
    >
      <p>
        I Intyg visas endast intyg som utfärdats digitalt till dig. Intyg har ingen ombudsfunktion och du kan inte ladda upp egna intyg.
      </p>
      <p>
        Om du saknar ett intyg, vänd dig till din mottagning.
      </p>
    </div>
  </div>
</div>
`;
